### ISSUE TITLE
Add Language Server Protocol (LSP) support for real-time linting

### ISSUE THREAD

**Issue by joshrotenberg** [joshrotenberg/mdbook-lint#18]
## Problem

Currently mdbook-lint requires manual execution or watch mode for feedback. Editors and IDEs would benefit from real-time linting feedback through LSP integration.

## Proposed Solution

Implement an LSP server that provides real-time markdown linting within editors supporting the Language Server Protocol.

### Implementation Areas

1. **LSP Server Binary**
   - New binary target: `mdbook-lint-lsp`
   - Implement LSP protocol handlers (initialize, textDocument/*, etc.)
   - Use existing linting engine for rule execution

2. **Core LSP Features**
   - **Diagnostics**: Real-time error/warning reporting as you type
   - **Code Actions**: Quick fixes for violations where possible
   - **Hover**: Show rule descriptions and documentation
   - **Document Formatting**: Auto-fix violations when possible

3. **mdbook-Specific Features**
   - **Workspace Awareness**: Understand book.toml and SUMMARY.md context
   - **Cross-File Validation**: Validate links across the entire book
   - **Include File Support**: Validate {{#include}} directives
   - **Configuration**: Use project's mdbook-lint configuration

### Technical Implementation

- Use `tower-lsp` crate for LSP protocol implementation
- Leverage existing `Document`, `Engine`, and rule infrastructure
- Support both file-based and in-memory document processing
- Handle incremental text changes for performance

### Use Cases

- **Real-time Feedback**: See violations as you type
- **Quick Fixes**: Apply suggested fixes with keyboard shortcuts
- **Rule Documentation**: Hover over violations to see rule explanations
- **Project Context**: Validate against entire book structure
- **Configuration Aware**: Respect project-specific lint settings

## Acceptance Criteria

- [ ] LSP server binary responds to standard LSP protocol
- [ ] Provides real-time diagnostics for markdown files
- [ ] Supports hover information for rules and violations
- [ ] Implements code actions for fixable violations
- [ ] Works with existing mdbook-lint configuration
- [ ] Handles workspace-wide validation (SUMMARY.md, cross-file links)
- [ ] Performance suitable for real-time editing
- [ ] Documentation for editor setup and configuration
- [ ] Tests cover LSP protocol compliance and functionality

## Future Enhancements

- Auto-completion for SUMMARY.md structure
- Semantic highlighting for mdbook syntax
- Refactoring support (rename files, update links)
- Integration with mdbook serve for live preview coordination

**Comment by joshrotenberg** [comment]
✅ **LSP Support Implemented in v0.4.0**

LSP server functionality has been successfully implemented with the following features:

## ✅ **Completed Features**
- **Real-time diagnostics** for markdown files as you type
- **Workspace awareness** - detects mdBook projects via  and 
- **Configuration loading** - automatically loads  project configs
- **Document sync** - handles file opening, changes, saves, and closes
- **Rule integration** - uses existing linting engine with all 63 rules (MD001-MD059 + MDBOOK001-007)

## 🛠️ **Technical Implementation**
- LSP server subcommand: `mdbook-lint lsp`
- Built with `tower-lsp` crate for protocol compliance  
- Supports both stdio and TCP modes
- Feature-gated dependencies (enabled by default)
- Full test coverage with integration tests

## 📚 **Documentation & Integration**
- Zed editor extension created and tested locally
- Ready for submission to Zed extensions registry
- API documentation included in mdBook docs

## 🎯 **Usage**
```bash
# Start LSP server
mdbook-lint lsp

# TCP mode for debugging  
mdbook-lint lsp --port 8080
```

The LSP server provides immediate feedback while editing, respects project configurations, and integrates seamlessly with the existing rule system. This significantly improves the developer experience for mdBook authors!
